### PR TITLE
Knot.x Launcher distribution

### PR DESCRIPTION
**Pull request by tomaszmichalak**
Knotx Launcher is a starting module for all Knot.x-based projects.

## Description
```
./gradlew clean publishToMavenLocal
unzip knotx-launcher-2.0.0-SNAPSHOT.zip
cd knotx-launcher-2.0.0-SNAPSHOT
./bin/start run-knotx
```

## Motivation and Context
 All modules in Knot.x can be replaced with custom implementation. This is the reason we deliver Knot.x Launcher as a separate ZIP distribution that can be reused in projects. Please note that Launcher module provides logger configuration too.

## Upgrade notes (if appropriate)
Logger configuration is moved from Server to Launcher module.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

**Review by malaskowski — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -64,20 +63,15 @@ dependencies {
...
 // Source sets
 // -----------------------------------------------------------------------------
-
-tasks.withType<JavaCompile>().configureEach {
-    with(options) {
-        sourceCompatibility = "1.8"
-        targetCompatibility = "1.8"
-        encoding = "UTF-8"
-    }
+tasks.named<JavaCompile>("compileJava") {
```
Comment (pun-ky) [inline build.gradle.kts:69]: wrap all tasks related into 

```
tasks {
   named(

   named(
}
```

File: build.gradle.kts [hunk 2]
```diff
@@ -125,16 +128,23 @@ tasks.register<Jar>("sourcesJar") {
...
+    }
+}
 tasks.register<Jar>("testJar") {
     from(sourceSets.named("junitTest").get().output)
     classifier = "tests"
 }
+tasks.register<Zip>("distZip") {
+    from("$buildDir/dist")
+}
+tasks.getByName("distZip").dependsOn("dist")
```
Comment (pun-ky) [inline build.gradle.kts:147]: avoid `getByName` it is eagerly creating task, see guidelines https://docs.gradle.org/current/userguide/task_configuration_avoidance.html

File: build.gradle.kts [hunk 3]
```diff
@@ -88,34 +82,43 @@ sourceSets.create("junitTest") {
...
+        from("conf")
+        into("$buildDir/dist/conf")
+    }
+    register<Copy>("dist") {
+        from(configurations.named("distConfig"), "$buildDir/libs")
+        include("*.jar")
+        exclude("*javadoc*", "*sources*", "*tests*")
+        into("$buildDir/dist/lib")
     }
+    getByName("distScript").dependsOn("build")
```
Comment (pun-ky) [inline build.gradle.kts:119]: use `named` / lazy API instead of `getByName`

File: build.gradle.kts [hunk 4]
```diff
@@ -88,34 +82,43 @@ sourceSets.create("junitTest") {
...
+            include("*.java")
+            filter<ReplaceTokens>("tokens" to tokens)
+        }
+        into("src/main/generated/io/knotx/launcher")
     }
     getByName<JavaCompile>("compileJava").dependsOn("templatesProcessing")
 
     named<RatTask>("rat") {
-        excludes.addAll("*.yml", "*.md", "**/*.md", "*.properties","**/build/*", "**/out/*", "**/generated/*", "gradle/wrapper/*", "gradlew", "gradlew.bat", ".idea/*")
+        excludes.addAll("*.yml", "*.md", "**/*.md", "*.properties", "script/*", "conf/*", "**/build/*", "gradle/wrapper/*", "gradlew", "gradlew.bat", ".idea/*")
```
Comment (pun-ky) [inline build.gradle.kts:101]: just `exclude(vararg)`

File: build.gradle.kts [hunk 5]
```diff
@@ -88,34 +82,43 @@ sourceSets.create("junitTest") {
...
-      into("src/main/generated/io/knotx/launcher")
+        val now = Date().time
+        val tokens = mapOf("project.version" to project.version, "build.timestamp" to "${now}")
+        inputs.properties(tokens)
+
+        from("src/main/java-templates") {
+            include("*.java")
+            filter<ReplaceTokens>("tokens" to tokens)
+        }
+        into("src/main/generated/io/knotx/launcher")
```
Comment (pun-ky) [inline build.gradle.kts:96]: you are here have one spec, but you can have multiple so that generally  more strict is `from { into } ` // nested / paired

**Pull request merged**